### ISSUE TITLE
Landrush prerequisite for libvirt provider

### ISSUE THREAD

**Issue by praveenkumar** [projectatomic/adb-atomic-developer-bundle#494]
Currently if a user do `vagrant up` using libvirt provider then it not able to ping because of libvirt also block `53` port so user actually to perform some steps before doing do `vagrant up` in this situation.

@LalatenduMohanty Please add the steps.

**Comment by LalatenduMohanty** [comment]
```
$ systemctl stop libvirtd
$ systemctl stop dnsmasq
```

Kill the dnsmasq process running on the host

```
$ ps aux | grep dns
$ kill -9 <pid>
$ ps aux | grep dns

```

Start the dnsmasq and libvirt process.

```
$ systemctl start dnsmasq
$ systemctl status dnsmasq
$ systemctl start libvirtd
$ systemctl status libvirtd
```

Now do the `vagrant up`

**Comment by LalatenduMohanty** [comment]
We should add a troubleshooting document for  Landrush and add this above information.

**Comment by ashetty1** [comment]
Also, disabling NetworkManager

**Comment by LalatenduMohanty** [comment]
@ashetty1 AFAIR we did not disable the NetworkManager in our setup i.e. in a CentOS7 host. Can you please explain little more about why you had to disable NetworkManager?

**Comment by bexelbie** [comment]
+1 for documentation.  @Preeticp and @thatdocslady can you all prepare a patch?

The issue is that libvirt starts dnsmasq before landrush can do so which blocks port 53.  This prevents the dnsmasq entries for landrush from being read.  The fix is to make sure that dnsmasq starts first.

In our install for libvirt, we should have the user start dnsmasq before libvirt

We should also have a note with the fix provided by @LalatenduMohanty for users who have already installed libvirt.

**Comment by mmalina** [comment]
I would like to add my own experience: I have seen a similar issue with mac and virtualbox - I did vagrant up and then I couldn't ping any of openshift.vm or cdk.vm - unknown host. But strangely, I was able to open the openshift console in my browser: https://openshift.cdk:8443/console
But I cannot reproduce it any longer.

**Comment by brgnepal** [comment]
@Preeticp @thatdocslady As discussed with @praveenkumar , we should document it like a warning for users Linux users who might have **libvirtd** running in background but want to use **VirtualBox** for CDK.

I am hit by this issue and hence proposing it.

In such case, we need to only following steps:

```
$ systemctl stop libvirtd
$ systemctl stop dnsmasq

# Kill the dnsmasq process running on the host

$ ps aux | grep dns
$ kill -9 <pid>
$ ps aux | grep dns

# Start the dnsmasq process only

$ systemctl start dnsmasq
$ systemctl status dnsmasq

# and

$ vagrant up
```

**Comment by surajssd** [comment]
@budhrg yes tried above steps on my fedora host (not in the ADB VM!) and it all works :)

**Comment by bexelbie** [comment]
should we recommend modifications to the stock /etc/dnsmasq.conf instead of this process?  The mods would allow for both to co-exist and not require this ordering to be maintained.

**Comment by dustymabe** [comment]
just taking a step back, I'd like to ask what use cases we are trying to solve with landrush? I think these are two of them:

1 - resloving guest virtual machine ip addr from the host
2 - resolving guestA ip addr from guestB

If we just care about number 1 then we can achieve the same effect by just modifying the configuration of dnsmasq:

```
$ cat /etc/dnsmasq.d/vagrant-landrush-cdk 
# Generated by landrush, a vagrant plugin
server=/cdk/127.0.0.1#10053
listen-address=127.0.0.100
bind-interfaces
```

This tells it to only listen on `127.0.0.100:53` and no other address/port combination. => no conflicts on bringup.

I think this solution would break the number 2 use case though as there won't be a dns server listening on the 192.168.122 addresses.

**Comment by dustymabe** [comment]
actually it looks like we have `config.landrush.guest_redirect_dns = false` so for the ADB/CDK it looks like we don't really care about use case number 2.

**Comment by bexelbie** [comment]
@dustymabe I think you are right, but I believe we actually have to edit /etc/dnsmasq.conf as well.  I believe there is a 0.0.0.0 bind in that file that is what is blocking the program from starting.

**Comment by dustymabe** [comment]
I don't think I had an issue when I used the `/etc/dnsmasq.d/vagrant-landrush-cdk` from my earlier comment. on Fedora 24

**Comment by bexelbie** [comment]
@dustymabe I suspect you are right and my brain misfired.

**Comment by bexelbie** [comment]
See and comment on https://github.com/vagrant-landrush/landrush/issues/258

**Comment by Preeticp** [comment]
Closed via #559

**Comment by LalatenduMohanty** [comment]
This is documented in the troubleshooting guide at  https://github.com/projectatomic/adb-atomic-developer-bundle/blob/master/docs/troubleshooting.adoc
